src/renderer/renderScene.ts

```diff
@@ -222,7 +222,7 @@ const renderLinearPointHandles = (
     appState,
   ).filter((midPoint) => midPoint !== null) as Point[];
 
-  if (!midPoints || !midPoints?.length) {
+  if (appState.editingLinearElement && !midPoints.length) {
     return;
   }
```

Comment:
we shouldn't return here without restoring the context (which I assumed caused the zoom bug). There shouldn't be much penalty for looping through an empty array (and it should happen in common case anyway?).